tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,66 @@
+import json
+import subprocess
+import urllib.request
+from time import sleep
+
+from tenacity import retry, stop_after_delay, wait_fixed
+
+
+@retry(wait=wait_fixed(5), stop=stop_after_delay(30))
+def test_dcgm_exporter():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "http://localhost:9400/metrics"
+
+    # Check the exporter endpoint, will raise an exception if the endpoint is not reachable
+    urllib.request.urlopen(endpoint)
+
+
+def test_dcgm_nv_hostengine():
+    """Check the dcgm-nv-hostengine service."""
+    nv_hostengine_service = "dcgm.nv-hostengine"
+
+    service = subprocess.run(
+        f"snap services {nv_hostengine_service}".split(),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+
+    assert " active" in service.stdout.strip(), f"{nv_hostengine_service} service is not active"
+
+
+def test_dcgmi():
+    """Test of the dcgmi command."""
+    result = subprocess.run(
+        "dcgm.dcgmi discovery -l".split(), check=True, capture_output=True, text=True
+    )
+    assert "GPU ID" in result.stdout.strip(), "DCGMI is not working"
```

Comment:
I think this assertion needs a comment, as it can be puzzling to see a test pass with a "not working" output. And it's odd that it shows that output in the CI, on my lxd it prints the following:
```
root@dcgm:~# dcgm.dcgmi discovery -l
0 GPUs found.
+--------+----------------------------------------------------------------------+
| GPU ID | Device Information                                                   |
+--------+----------------------------------------------------------------------+
+--------+----------------------------------------------------------------------+
0 NvSwitches found.
+-----------+
| Switch ID |
+-----------+
+-----------+
0 CPUs found.
+--------+----------------------------------------------------------------------+
| CPU ID | Device Information                                                   |
+--------+----------------------------------------------------------------------+
+--------+----------------------------------------------------------------------+
```

Reply:
also in a noble vm, although oddly the output is different in a jammy vm:
```
root@jammyvm:~# dcgm.dcgmi discovery -l
0 GPUs found.
+--------+----------------------------------------------------------------------+
| GPU ID | Device Information                                                   |
+--------+----------------------------------------------------------------------+
+--------+----------------------------------------------------------------------+
0 NvSwitches found.
+-----------+
| Switch ID |
+-----------+
+-----------+
Error: Cannot get devices from remote node. Return: Host engine connection invalid/disconnected
Cannot get CPU list from remote node. Error: -21: Host engine connection invalid/disconnected
```

Reply:
@aieri Does it return 0 or something else when there is an error in jammy?

Reply:
it returns 235